### ISSUE TITLE
Is there a bug with timezone?

### ISSUE THREAD

**Issue by ahmedsajid** [blue-yonder/bonfire#36]
The README file suggests that there's still a bug but the Closed issue #25 suggests otherwise.

My local machine time and Graylog Server both are in `America/Toronto` timezone and I'm getting some issues with incorrect time range.

It will be great if someone could clarify.

Also, this tool is amazing!

**Comment by ahmedsajid** [comment]
(adding more info)

Here's the query I'm running:
`bonfire -h 'graylog.example.com' -p --port 443 --tls --endpoint '/api' -u Ahmed.Sajid  -r 000000000000000000000001  -f `

On server side (Nginx access logs)
```
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:57:58 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A57%3A52.825&to=2021-08-09+17%3A57%3A55.022 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:01 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A57%3A55.022&to=2021-08-09+17%3A57%3A56.736 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:03 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A57%3A56.736&to=2021-08-09+17%3A57%3A59.385 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:06 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A57%3A59.385&to=2021-08-09+17%3A58%3A01.595 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:08 -0400] "GET /api/search/universal/absolute?query=%2A&limit=1&filter=streams%3A000000000000000000000001&from=2021-08-09+21%3A58%3A01.595&to=2021-08-09+13%3A58%3A04.287 HTTP/1.1" 200 2204 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:14 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A58%3A06.923&to=2021-08-09+17%3A58%3A09.259 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:19 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A58%3A12.355&to=2021-08-09+17%3A58%3A14.696 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:21 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A58%3A14.696&to=2021-08-09+17%3A58%3A17.319 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:28 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A58%3A21.467&to=2021-08-09+17%3A58%3A24.353 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:30 -0400] "GET /api/search/universal/absolute?query=%2A&limit=1&filter=streams%3A000000000000000000000001&from=2021-08-09+21%3A58%3A24.353&to=2021-08-09+13%3A58%3A26.918 HTTP/1.1" 200 2204 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:32 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A58%3A26.918&to=2021-08-09+17%3A58%3A28.834 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:34 -0400] "GET /api/search/universal/absolute?query=%2A&limit=1&filter=streams%3A000000000000000000000001&from=2021-08-09+21%3A58%3A28.834&to=2021-08-09+13%3A58%3A30.750 HTTP/1.1" 200 2204 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:37 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A58%3A30.750&to=2021-08-09+17%3A58%3A33.327 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
10.0.10.1 - Ahmed.Sajid [09/Aug/2021:09:58:39 -0400] "GET /api/search/universal/absolute?query=%2A&limit=0&filter=streams%3A000000000000000000000001&from=2021-08-10+01%3A58%3A33.327&to=2021-08-09+17%3A58%3A35.656 HTTP/1.1" 200 2206 "-" "python-requests/2.25.1"
```

Why does the time range changes every few calls?

```
from=2021-08-10 01:57:52.825&to=2021-08-09 17:57:55.022
from=2021-08-10 01:57:55.022&to=2021-08-09 17:57:56.736
from=2021-08-10 01:57:56.736&to=2021-08-09 17:57:59.385
from=2021-08-10 01:57:59.385&to=2021-08-09 17:58:01.595
from=2021-08-09 21:58:01.595&to=2021-08-09 13:58:04.287
from=2021-08-10 01:58:06.923&to=2021-08-09 17:58:09.259
from=2021-08-10 01:58:12.355&to=2021-08-09 17:58:14.696
from=2021-08-10 01:58:14.696&to=2021-08-09 17:58:17.319
from=2021-08-10 01:58:21.467&to=2021-08-09 17:58:24.353
from=2021-08-09 21:58:24.353&to=2021-08-09 13:58:26.918
from=2021-08-10 01:58:26.918&to=2021-08-09 17:58:28.834
from=2021-08-09 21:58:28.834&to=2021-08-09 13:58:30.750
from=2021-08-10 01:58:30.750&to=2021-08-09 17:58:33.327
from=2021-08-10 01:58:33.327&to=2021-08-09 17:58:35.656
```

I'll keep digging and will report back if I find anything.

**Comment by valentin-krasontovitsch** [comment]
sorry about the slow response, i believe this is a dupliate of #35 - and thank you for providing a very good looking solution! please confer my review of your pull request.

regarding your question as to why the time range changes - follow mode is implemented in such a way that it keeps querying the server for new logs, changing the time range for the logs every query, so you get all logs within round about 3 seconds from time a, and then the logs from the next 3 seconds, and then the following 3 seconds, and so on.

does that make sense?

**Comment by ahmedsajid** [comment]
Thanks for getting back.

Yes it does makes sense.

I'll spend sometime and address your comments on the PR.

**Comment by valentin-krasontovitsch** [comment]
fixed by 21cdfc27de2e3d9a628d1d5526e603e871ac99a5
